### DISCUSSION TITLE
On overriding userdir with Steam directory

### DISCUSSION THREAD

**Discussion by 4LT** [andrei-drexler/ironwail#117] [Ideas]
Maybe this is just my personal preference, but I only use my Steam Quake installation to copy out files I need.  I prefer to keep my mods in `~/.ironwail`, so I find it annoying when the userdir is overwritten with my Steam Quake directory, requiring me to run ironwail with `-basedir ~/.ironwail`.  I thought I'd raise this as discussion instead of an issue, as maybe I'm the odd one out here.

(This is when compiling with `DO_USERDIRS=1` ofc)

**Comment by andrei-drexler** [comment]
The userdir should only be overridden when game data can't be found in the current directory (or its parent), a situation which would've resulted in an error anyway before Steam detection was added. So if you're running the engine just like before (or like any other engine), everything should work as it used to (if not, that's a bug).

**Comment by 4LT** [comment]
> The userdir should only be overridden when game data can't be found in the current directory (or its parent), a situation which would've resulted in an error anyway before Steam detection was added.

Oh! I didn't realize, that's divergent behavior from QuakeSpasm.

I can run QuakeSpasm from any directory so long as I have a pak0.pak in `~/.quakespasm/id1`

**Reply by andrei-drexler** [reply]
Ah, that's not a use case I expected. https://github.com/andrei-drexler/ironwail/commit/aacf9c61f1a785749fea5542dbc4f61f2dce189a should fix it, but I haven't actually rebooted into Ubuntu to test it.

**Reply by 4LT** [reply]
I'll give it a go when I'm out of work

**Reply by 4LT** [reply]
Alright, it appears to be working as expected, I can launch ironwail from anywhere and access all my mods

**Reply by andrei-drexler** [reply]
Great, thanks for letting me know!
